core/dbt/contracts/graph/model_config.py

```diff
@@ -460,6 +461,20 @@ class NodeConfig(NodeAndTestConfig):
     grants: Dict[str, Any] = field(
         default_factory=dict, metadata=MergeBehavior.DictKeyAppend.meta()
     )
+    docs: Dict[str, Any] = field(
+        default_factory=dict,
+        metadata=MergeBehavior.Update.meta(),
+    )
+
+    # we validate that node_color has a suitable value to prevent dbt-docs from crashing
+    def __post_init__(self):
```

Comment:
@stu-k How do you feel about implementing color validation at the dbt-core layer vs. the dbt-docs layer?

https://github.com/dbt-labs/dbt-docs/pull/279#pullrequestreview-1010780959

With that in mind, do we need to hard code that list of colors based on your comment above?

Reply:
Hey Stu!

Makes sense to me. We can go ahead and include this defensive code in the `dbt-docs` codebase. 

Part of our thinking was to make sure dbt tells the developer that they have the wrong colors when compiling their code. I know a lot of dbt users that don't run `dbt docs generate` until it hits a scheduled job. Therefore, the troubleshooting happens on the tailend vs. upfront. But sounds like we're okay with that tradeoff given maintainability of the codebase matters more.

Reply:
Thanks both for sharing your perspectives! I see compelling points on both sides, and I could go either way.

**Truth:** There's really zero other CSS/HTML/JS in `dbt-core`. `dbt-core` does know a _teensy_ bit about Markdown, just enough to make `description` fields populate correctly in `dbt-docs`, but it doesn't do any of the rendering itself.

Also, users _should_ include docs generation + manual inspection of docs during development / CI, if they're messing with custom node colors, to makes sure that what they're doing doesn't break their docs site. (I bet we can _also_ add some defensive logic to the docs site that ignores invalid color codes, rather than crashing.)

At the same time — by adding `docs.node_color` as a first-class config/property in `dbt-core`, similar to `docs.show` — we are setting an expectation that `dbt-core` knows about it and takes an interest in validating this config, more than it would for "any" fields like `meta`. That, along with making this configurable according to file paths (FQNs) and other rule sets, is one of the reasons I'd be supportive of incorporating this logic into `dbt-core` directly.

Reply:
Having said my piece above, I'm happy to defer to @stu-k's preference, motivated by separation of concerns between `dbt-core` and `dbt-docs`